torch/_vmap_internals.py

```diff
@@ -34,19 +34,65 @@
     'output (got {num_outputs} outputs) of {fn}.'
 )
 
+EXPECTED_IN_DIMS_TO_BE_INT_OR_TUPLE = (
+    'vmap({fn}, in_dims={in_dims}, ...): expected `in_dims` to be int or tuple, '
+    'got: {actual_type}.'
+)
```

Comment:
No action needed on this comment: I personally prefer having messages inline at their use sites, if they're only used once. Makes it easier to see what the error message is and ensure that the format string is up to date :) (also, you can't use f-strings in this style!)

Reply:
Did we drop support for Python < 3.6? (I know we dropped support for Python 2, but I didn't realize the < 3.6 part)

I agree with your comment, reading 66 lines of error messages at the top of the file and away from the callsites makes me sad. Will fix in a follow-up.